common/priorityQueue.py

```diff
@@ -0,0 +1,84 @@
+import bisect
+
+class PriorityQueue:
+    def __init__(self):
+        self._queue = []
+        self._size = 0
+        self._id = 0
+
+        # a unique sender is determined by the path to the sender's emails
+        self._uniqueSenders = set([])
+
+    def getLength(self):
+        return self._size
+
+    # returns the popped item
```

Comment:
This code comment is missing some context.  push() is popping an item?  Maybe the first comment should describe what push() does.  Also you don't seem to use the return value, so I'm not sure why you bother returning what was popped anyway.